packages/frontend/src/Pages/PersonalPage/PersonalPage.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { Button } from "@mui/material";
+import { AppRoutes } from "../../constants/app-routes.constants";
+import { signOutCustomer } from "../../app/storeSlice";
+import { useAppSelector, useAppDispatch } from "../../app/hooks";
+import Purchases from "../../components/Purchases/Purchases";
+import "./PersonalPage.scss";
+
+const PersonalPage: React.FC = () => {
+  const {
+    customer: { name, lastName, email, address },
+    isAuthorized,
+    isAdmin,
+  } = useAppSelector((store) => store.data);
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (!isAuthorized) navigate(AppRoutes.MAIN, { replace: true });
```

Comment:
I'm thinking about this line of code and to my mind we better to configure router at App.tsx. Like if user not authorized, then he simply can't enter this route (PersonalPage). Imagine we have 20 pages when user need to be authorized and we will have 20 useEffects only to cover redirect in case user not authorized

Reply:
done